tests/shared/src/test/scala/scala/meta/tests/tokenizers/TokenizerCoverageSuite.scala

```diff
@@ -0,0 +1,60 @@
+package scala.meta.tests.tokenizers
+
+import scala.meta._
+
+class TokenizerCoverageSuite() extends BaseTokenizerCoverageSuite {
+
+  // Term
+  check[Term.Annotate]("→(a)←: →@A←")
+  check[Term.Apply]("→(f)←(→(((a)))←)")
+  check[Term.Apply]("→(f)←(→(a)←)")
+  check[Term.Apply]("→(f)←(→{ case a => a }←)")
+  check[Term.Apply]("→(f)←(→{ x }←)")
+  check[Term.ApplyInfix]("→(a)← →op← →(b)←")
+  check[Term.ApplyInfix]("→(a, b)← →op← (→c←, →d←)")
+  check[Term.ApplyInfix]("→(a, b)← →op← →((c, d))←")
+  check[Term.ApplyInfix]("→1← →+← →1←")
+  check[Term.ApplyInfix]("→a← →f← →(b)←")
+  check[Term.ApplyType]("→(f)← [→A←,→B←]")
+  check[Term.ApplyType]("→(f)← [→A←]")
+  check[Term.ApplyUnary]("→-← →(a)←")
```

Comment:
Is this expected that the parentheses belong to `Term.Name`?

Reply:
There seems to be several related places where I would expect redundant parentheses to belong to the parent node.

Reply:
@olafurpg Right now, my focus is to describe the current state. I will do a followup with the positions we expect.

This is the pattern I saw in the parser for lists:

* if there is a single element, it will capture the parenthesis (ex: `a f (b)`).
* If there is no element, it will not capture the parenthesis (ex `a f ()`)
* if there is more than one element, it will not capture the parenthesis (ex `a f (b, c)`)

For structures (like if, while, etc) it will never capture parenthesis when required:

* `if (p) t` gives `p`
* `return (a)` gives `(a)`

one exception is classes:

* `class A(b: B)` gives `(b: B)`

Reply:
That's a good strategy!